### ISSUE TITLE
make parent frame check origin of child frame when receiving message

### ISSUE THREAD

**Issue by mykmelez** [andreasgal/j2me.js#909]
To avoid the possibility that the child frame will somehow be navigated to a malicious page that abuses the APIs it accesses through the parent frame, the parent frame should check the origin of the child frame to ensure it is as expected when it receives a message from the child frame.

**Comment by mozfreddyb** [comment]
Ideally, this would happen in index.js, `DumpPipe.handleEvent`, checking that `event.target.location` is the URL of main.html

**Comment by mozfreddyb** [comment]
The patch in #977 checks that the frame still contains to `main.html`, but it doesn't go deeper: While it's rather unlikely that an attacker could get the frame to load further frames, we might want to secure check that a message does not come from a nested frame.

From observing what `mozbrowsershowmodalprompt` events contain, I noticed that `currentTarget`, `target`as well as `originalTarget` *all* point to the mozbrowser frame and not the nested frame :/

The only thing I found meaningful was `detail.title`, which is `"The page at app://….mozilla.org says:"`.
The origin in there seems reliable, but it also feels like a very dirty hack.
